benchmark/Streamly/Benchmark/Data/Array/Stream/Foreign.hs

```diff
@@ -192,10 +192,14 @@ moduleName = "Data.Array.Stream.Foreign"
 
 main :: IO ()
 main = do
-    (value, cfg, benches) <- parseCLIOpts defaultStreamSize
     env <- mkHandleBenchEnv
-    value `seq` runMode (mode cfg) cfg benches
-        (allBenchmarks env)
+#if TASTY
+    (value, ingredients) <- parseCLIOpts defaultStreamSize $ bgroup "All" (allBenchmarks env)    
+    value `seq` defaultMainWithIngredients ingredients $ bgroup "All" (allBenchmarks env)
+#else
+    (value, cfg, benches) <- parseCLIOpts defaultStreamSize
+    value `seq` runMode (mode cfg) cfg benches (allBenchmarks env)
+#endif   
```

Comment:
Define allBenchmarks as `allBenchmarks env _ =` and call `runWithCLIOpts (return . allBenchmarks env)`.